modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -578,6 +586,17 @@ public class IgniteImpl implements Ignite {
 
         indexManager = new IndexManager(schemaManager, distributedTblMgr, catalogManager, metaStorageMgr, registry);
 
+        indexBuilder = new IndexBuilder(name, Runtime.getRuntime().availableProcessors(), replicaSvc);
+
+        indexBuildController = new IndexBuildController(
+                indexBuilder,
```

Comment:
Ok, I guess no one, except for the controller, uses index builder. But, we accept the builder as a dependency, because it simplifies tests?
On one hand, I'd love to get rid of the constructor parameter, but on the other, your approach is more flexible. This comment is just my thoughts, you don't need to change anything

Reply:
ok